packages/gemini-runner/src/GeminiRunner.ts

```diff
@@ -0,0 +1,561 @@
+			this.emit("message", message);
+		}
+	}
+
+	/**
+	 * Stop the current Gemini session
+	 */
+	stop(): void {
+		if (this.process) {
+			console.log("[GeminiRunner] Stopping Gemini process");
+			this.process.kill("SIGTERM");
+			this.process = null;
+		}
+
+		if (this.sessionInfo) {
+			this.sessionInfo.isRunning = false;
+		}
+
+		// Complete streaming prompt if active
+		if (this.streamingPrompt) {
+			this.streamingPrompt.complete();
+		}
+	}
+
+	/**
+	 * Check if the session is currently running
+	 */
+	isRunning(): boolean {
+		return this.sessionInfo?.isRunning ?? false;
+	}
+
+	/**
+	 * Get all messages from the current session
+	 */
+	getMessages(): SDKMessage[] {
+		return [...this.messages];
+	}
+
+	/**
+	 * Set up logging streams for this session
+	 */
+	private setupLogging(): void {
+		const logsDir = join(this.cyrusHome, "logs");
+		const workspaceName =
+			this.config.workspaceName ||
+			(this.config.workingDirectory
+				? this.config.workingDirectory.split("/").pop()
+				: "default") ||
+			"default";
+		const workspaceLogsDir = join(logsDir, workspaceName);
+		const sessionId = this.sessionInfo?.sessionId || "pending";
+
+		// Close existing streams if they exist
+		if (this.logStream) {
+			this.logStream.end();
+		}
+		if (this.readableLogStream) {
+			this.readableLogStream.end();
+		}
+
+		// Ensure logs directory exists
+		mkdirSync(workspaceLogsDir, { recursive: true });
+
+		// Create log streams
+		const logPath = join(workspaceLogsDir, `${sessionId}.ndjson`);
+		const readableLogPath = join(workspaceLogsDir, `${sessionId}.log`);
+
+		console.log(`[GeminiRunner] Logging to: ${logPath}`);
+		console.log(`[GeminiRunner] Readable log: ${readableLogPath}`);
+
+		this.logStream = createWriteStream(logPath, { flags: "a" });
+		this.readableLogStream = createWriteStream(readableLogPath, { flags: "a" });
+
+		// Log session start
+		const startEntry = {
+			type: "session-start",
+			sessionId,
+			timestamp: new Date().toISOString(),
+			config: {
+				model: this.config.model,
+				workingDirectory: this.config.workingDirectory,
+			},
+		};
+		this.logStream.write(`${JSON.stringify(startEntry)}\n`);
+		this.readableLogStream.write(
+			`=== Session ${sessionId} started at ${new Date().toISOString()} ===\n\n`,
+		);
+	}
+
+	/**
+	 * Write a human-readable log entry for a message
+	 */
+	private writeReadableLogEntry(message: SDKMessage): void {
+		if (!this.readableLogStream) return;
+
+		const timestamp = new Date().toISOString();
+		this.readableLogStream.write(`[${timestamp}] ${message.type}\n`);
+
+		if (message.type === "user" || message.type === "assistant") {
+			const content =
+				typeof message.message.content === "string"
+					? message.message.content
+					: JSON.stringify(message.message.content, null, 2);
+			this.readableLogStream.write(`${content}\n\n`);
+		} else {
+			// Other message types (system, result, etc.)
+			this.readableLogStream.write(`${JSON.stringify(message, null, 2)}\n\n`);
+		}
+	}
```

Comment:
we should consolidate the approach to logging. Making it so that sure we can see differentiated which runner it was, but other than that the logging for different runners should operate consistently in terms of the underlying